pkg/controller/astarte/reconcile/astarte_generic_api.go

```diff
@@ -142,6 +137,36 @@ func EnsureAstarteGenericAPIWithCustomProbe(cr *apiv1alpha1.Astarte, api apiv1al
 	return nil
 }
 
+func checkShouldDeployAPI(reqLogger logr.Logger, deploymentName string, cr *apiv1alpha1.Astarte, api apiv1alpha1.AstarteGenericAPISpec,
+	component apiv1alpha1.AstarteComponent, c client.Client) bool {
+	if !pointy.BoolValue(api.Deploy, true) {
+		reqLogger.V(1).Info("Skipping Astarte Component Deployment")
+		// Before returning - check if we shall clean up the Deployment.
+		// It is the only thing actually requiring resources, the rest will be cleaned up eventually when the
+		// Astarte resource is deleted.
+		theDeployment := &appsv1.Deployment{}
+		if err := c.Get(context.TODO(), types.NamespacedName{Name: deploymentName, Namespace: cr.Namespace}, theDeployment); err == nil {
+			reqLogger.Info("Deleting previously existing Component Deployment, which is no longer needed")
+			if err = c.Delete(context.TODO(), theDeployment); err != nil {
+				reqLogger.Error(err, "Could not delete previously existing Component Deployment")
+			}
+		}
+
+		// In any case, we should not deploy
+		return false
+	}
+
+	// If we do need to deploy, check any constraints
+	version := getSemanticVersionForAstarteComponent(cr, api.Version)
+	constraint, _ := semver.NewConstraint(">= 1.0.0")
```

Comment:
Will this work also for 1.0.0-rc? I assume we will test with that before the final release

Reply:
The semantic version check always ensures that no prerelease is ever considered in the checks: https://github.com/astarte-platform/astarte-kubernetes-operator/blob/b8d8950663ddd69112933bf969bffa7eda59b9db/pkg/controller/astarte/reconcile/utils.go#L376

As such yes, any 1.0.0 prerelease will pass that constraint.